stl/inc/format

```diff
@@ -2428,6 +2449,45 @@ using _Fmt_wit = back_insert_iterator<_Fmt_buffer<wchar_t>>;
 _EXPORT_STD using format_context  = basic_format_context<_Fmt_it, char>;
 _EXPORT_STD using wformat_context = basic_format_context<_Fmt_wit, wchar_t>;
 
+#if _HAS_CXX23
+_EXPORT_STD enum class range_format { disabled, map, set, sequence, string, debug_string };
+
+template <class _Ty>
+struct _Invalid_format_kind {
+    static_assert(_Always_false<_Ty>, "A program that instantiates the primary template of format_kind is ill-formed. "
+                                      "(N4964 [format.range.fmtkind]/1)");
+};
+
+_EXPORT_STD template <class _Ty>
+constexpr _Invalid_format_kind<_Ty> format_kind{};
+
+template <class _Ty>
+inline constexpr bool _Is_two_tuple = false;
+
+template <class _Ty, class _Uty>
+inline constexpr bool _Is_two_tuple<pair<_Ty, _Uty>> = true;
+
+template <class _Ty, class _Uty>
+inline constexpr bool _Is_two_tuple<tuple<_Ty, _Uty>> = true;
+
+template <_RANGES input_range _Rng>
+    requires same_as<_Rng, remove_cvref_t<_Rng>>
+constexpr range_format format_kind<_Rng> = []() consteval {

```

Comment:
Does this need to be `inline` (as are the `_Is_two_tuple` variable templates) for MSVC to give it external linkage, or did that issue get fixed while I wasn't looking?

Reply:
Oh, that's a good catch!

Reply:
Ditto followup.

Reply:
Should we mention CWG-2387 (seems not yet implemented by MSVC)?

Reply:
> Should we mention [CWG-2387](https://cplusplus.github.io/CWG/issues/2387.html) (seems not yet implemented by MSVC)?

If we mention it only here then it's an inconsistency with every other `inline` variable template. If anything, we should have an issue reminding us to audit and remove all the extraneous `inline`s when our supported compilers are all ready.